src/components/InteractiveMap/Escher.vue

```diff
@@ -0,0 +1,143 @@
+      fluid
+      fill-height
+      class="overlay"
+      v-if="initializingEscher !== null || isLoadingMap"
+    >
+      <v-layout align-center justify-center>
+        <v-progress-circular
+          indeterminate
+          size="40"
+          :width="2"
+          class="mr-2"
+          color="white"
+        ></v-progress-circular>
+        <p class="display-1 white--text mb-0">
+          <span v-if="initializingEscher !== null">Initializing Escher...</span>
+          <span v-else-if="isLoadingMap">Loading map...</span>
+        </p>
+      </v-layout>
+    </v-container>
+    <div ref="escher" class="fill-height"></div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+/// <reference path="@/types/escher.d.ts" />
+import * as escher from "@dd-decaf/escher";
+
+export default Vue.extend({
+  name: "Escher",
+  props: ["mapData", "fluxDistribution"],
+  data: () => ({
+    escherBuilder: null,
+    initializingEscher: null,
+    isLoadingMap: false
+  }),
+  watch: {
+    mapData(value) {
+      const loadMap = () => {
+        this.isLoadingMap = true;
+        // Delay the actual loading of the map to give Vue time to update the
+        // interface (after setting isLoadingMap to true). Otherwise, Escher
+        // would freeze execution until it's complete and the user would never
+        // actually see the loading message.
+        setTimeout(() => {
+          this.escherBuilder.load_map(value);
+          this.isLoadingMap = false;
+        }, 10);
+      };
+
+      // Wait for escher to initialize before loading the map.
+      if (this.initializingEscher !== null) {
+        this.initializingEscher.then(loadMap);
+      } else {
+        loadMap();
+      }
+    },
+    fluxDistribution(value) {
+      this.escherBuilder.set_reaction_data(value);
+      this.escherBuilder._update_data(true, true);
+    }
+  },
+  methods: {
+    reactionState(id: string, type?: string) {
+      // TODO
+    },
+    handleKnockout(reactionId: string) {
+      // TODO
+    },
+    handleKnockoutGenes(reactionId: string) {
+      // TODO
+    },
+    handleSetAsObjective(reactionId: string) {
+      // TODO
+    },
+    handleChangeBounds(reactionId: string, lower: string, upper: string) {
+      // TODO
+    },
+    handleResetBounds(reactionId: string) {
+      // TODO
+    },
+    handleObjectiveDirection(reactionId: string) {
+      // TODO
+    }
+  },
+  mounted() {
+    this.initializingEscher = new Promise((resolve, reject) => {
+      this.escherBuilder = escher.Builder(null, null, null, this.$refs.escher, {
+        menu: "zoom",
+        scroll_behavior: "zoom",
+        fill_screen: false,
+        ignore_bootstrap: true,
+        never_ask_before_quit: true,
+        reaction_styles: ["color", "size", "text", "abs"],
+        identifiers_on_map: "bigg_id",
+        hide_all_labels: false,
+        hide_secondary_metabolites: false,
+        highlight_missing: true,
+        reaction_scale: [
```

Comment:
I will create an issue about this. Zak has updated Escher (I don't know if our fork has those changes) such that we can have more control over the scale. For example, we can set the same scale across multiple cards.

Reply:
Sounds good